src/Compilers/CSharp/Test/Symbol/Symbols/Retargeting/NoPia.cs

```diff
@@ -1367,6 +1367,227 @@ public void GenericsClosedOverLocalTypes1_1()
+        public ValueTuple(T1 item1, T2 item2) { }
+    }
+}
+
+[CompilerGenerated, TypeIdentifier(""f9c2d51d-4f44-45f0-9eda-c9d599b58257"", ""S1"")]
+public struct S1 { }
+
+public class C
+{
+    public ValueTuple<S1, S1> Test1()
+    {
+        throw new Exception();
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, options: TestOptions.ReleaseDll, assemblyName: "comp");
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp);
+
+            var assemblies1 = MetadataTestHelpers.GetSymbolsForReferences(
+                                new CSharpCompilation[] { comp },
+                                null,
+                                new MetadataReference[] { },
+                                null);
+
+            Assert.Equal(SymbolKind.ErrorType, assemblies1[0].GlobalNamespace.GetMember<MethodSymbol>("C.Test1").ReturnType.Kind);
+
+            var assemblies2 = MetadataTestHelpers.GetSymbolsForReferences(
+                                new CSharpCompilation[] { },
+                                null,
+                                new MetadataReference[] { comp.ToMetadataReference() },
+                                null);
+
+            Assert.Equal(SymbolKind.ErrorType, assemblies2[0].GlobalNamespace.GetMember<MethodSymbol>("C.Test1").ReturnType.Kind);
+        }
+
+        [ClrOnlyFact]
+        public void ImportedValueTuple()
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    [CompilerGenerated, TypeIdentifier(""f9c2d51d-4f44-45f0-9eda-c9d599b58257"", ""ValueTuple"")]
+    public struct ValueTuple<T1, T2>
+    {
+        public ValueTuple(T1 item1, T2 item2) { }
+    }
+}
+
+public class C
+{
+    public ValueTuple<int, int> Test1()
+    {
+        throw new Exception();
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, options: TestOptions.ReleaseDll, assemblyName: "comp");
+            comp.VerifyDiagnostics();
+
+            var assemblies1 = MetadataTestHelpers.GetSymbolsForReferences(
+                                new CSharpCompilation[] { comp },
+                                null,
+                                new MetadataReference[] { },
+                                null);
+
+            Assert.Equal(SymbolKind.ErrorType, assemblies1[0].GlobalNamespace.GetMember<MethodSymbol>("C.Test1").ReturnType.Kind);
+
+            var assemblies2 = MetadataTestHelpers.GetSymbolsForReferences(
+                                new CSharpCompilation[] { },
+                                null,
+                                new MetadataReference[] { comp.ToMetadataReference() },
+                                null);
+
+            Assert.Equal(SymbolKind.ErrorType, assemblies2[0].GlobalNamespace.GetMember<MethodSymbol>("C.Test1").ReturnType.Kind);
+        }
+
+        [ClrOnlyFact]
+        public void CannotEmbedValueTuple()
```

Comment:
Added test for tuple literal and for deconstruction (this will need to merge with PR for deconstruction return-type).
I'll stop by to understand the last part of your comment.